hikari/internal/reflect.py

```diff
@@ -56,6 +57,9 @@ def resolve_signature(func: typing.Callable[..., typing.Any]) -> inspect.Signatu
         A `inspect.Signature` object with all forward reference annotations
         resolved.
     """
+    if sys.version_info >= (3, 10):
```

Comment:
Would it be bette to have this check outside?

Reply:
Either approach works, this approach just in this case this is just kinda easier since avoids stuff like having to duplicate the docstring between the two definitions